### PR TITLE
PE-1580: Survey Feedback Modal

### PR DESCRIPTION
**Pull request by matibat**
Thew integration branch of the sub-tasks.

### CODE REVIEW

File: lib/components/create_manifest_form.dart [hunk 1]
```diff
@@ -37,7 +38,7 @@ class CreateManifestForm extends StatelessWidget {
   @override
   Widget build(BuildContext context) =>
       BlocConsumer<CreateManifestCubit, CreateManifestState>(
-          listener: (context, state) {
+          listener: (context, state) async {
```
Comment (fedellen) [inline lib/components/create_manifest_form.dart:41]: Is this still async?

Comment (karlprieb) [reply lib/components/create_manifest_form.dart:41]: good catch @fedellen I believe @matibat just forgot to clean this up

File: lib/components/feedback_survey.dart [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    );
+
+class FeedbackSurveyModal extends StatelessWidget {
+  const FeedbackSurveyModal({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) =>
+      BlocBuilder<FeedbackSurveyCubit, FeedbackSurveyState>(
+        builder: (context, state) => Stack(
+          children: [
```
Comment (fedellen) [inline lib/components/feedback_survey.dart:26]: I wonder if we could break some if this up here into individual widgets with some descriptive names.

File: lib/l10n/app_en.arb [hunk 3]
```diff
@@ -1291,5 +1291,33 @@
...
+  "@leaveFeedback": {
+    "description": "The action of leaving feedback"
+  },
+  "noThanks": "No thanks! Don't remind me again",
+  "@noThanks": {
+    "description": "Don't leave feedback, and don't ask again for it"
+  },
+  "weWontRemindYou": "Ok cool! We won't remind you again.",
+  "@weWontRemindYou": {
+    "description": "We know the user don't want to leave feedback ATM"
```
Comment (fedellen) [inline lib/l10n/app_en.arb:1313]:
```suggestion
    "description": "We let the user know that we won't remind them to leave feedback again"
```

File: lib/components/feedback_survey.dart (hunk 2 above)
Comment (matibat) [reply lib/components/feedback_survey.dart:26]: I actually would love to split all of the widgets into individual reusable components. I think we don't yet have a place in the repo to place them...

File: lib/components/create_manifest_form.dart (hunk 1 above)
Comment (matibat) [reply lib/components/create_manifest_form.dart]: Nice catch bbe927425a72de7430a16306c32165426d294638

File: lib/components/feedback_survey.dart (hunk 2 above)
Comment (karlprieb) [reply lib/components/feedback_survey.dart:26]: @thiagocarvalhodev correct me if I'm wrong, but I feel that's the way flutter developers are used to do things.
I really dislike, but all our code is like that right now. We should invest some time creating more individual widgets when we start to work on the redesign.

**Review by karlprieb — DISMISSED** [review]

**Review by karlprieb — DISMISSED** [review]

File: lib/components/feedback_survey.dart (hunk 2 above)
Comment (thiagocarvalhodev) [reply lib/components/feedback_survey.dart:26]: @karlprieb , is not. Most of the projects that I worked we used our component library to build the UI, based on our design system. I mean, we had our own buttons, texts, inputs, etc...

I was talking with Lucas about our own design system, and it is one of our goals on the UI side, but we won't have it on time to implement the redesign. So, I strongly recommend building our component library too, but starting smaller and simple. We can start creating Buttons, Texts, Inputs, Modals, etc... every new component from the redesign should be on that library, while we don't have our design system well defined on the UI side. After that, we can build a true design system, integrating UI/UX with devs.

We also must be very critical of the new components' implementation.

**Review by karlprieb — DISMISSED** [review]

**Review by jdaev — APPROVED** [review]

**Pull request merged**